### DISCUSSION TITLE
Issues accessing "msdos" type etherDFS share

### DISCUSSION THREAD

**Discussion by evan** [retronas/discussions#13] [Ideas]
I set up samba and an msdos-type mount point called "dos" according to the directions: https://github.com/danmons/retronas/wiki/FAT

- From Windows 98 (hardwired), the samba share is discoverable as `retrosmb` and the `dos` folder is accessible
- From Windows 10 (wifi), the samba share is not discoverable, but it can be accessed by name (both `retronas` and `retrosmb`), and the `dos` folder is accessible
- From OS X (wifi), the samba share is discoverable as `retrosmb`, but accessing the `dos` folder fails with an error: `The folder “dos” can’t be opened because you don’t have permission to see its contents.`

I'm not sure what settings if any to change on the server or client to allow OS X access to the `dos` folder. 

It would also be nice if the share was discoverable on Windows 10 and had a consistent name (not sure why the netbios name in smb.conf is different). I am running RetroNAS within a VM in Unraid, if it matters.

**Comment by evan** [comment]
Turning on "SMB 1.0/CIFS File Sharing Support" in the Windows 10 Add/Remove Optional Features seems to make it discover the share, at least.

I also see the note about the netbios name being different to prevent the netatalk conflict.

**Comment by sairuk** [comment]
enabling smb 1.0 support on win10 is not the best idea, it is fine to have it enabled on linux as it doesn't suffer from the same issues as windows.

I took a look at our doco on the supported clients and access methods for is each is OK and discoverability is a nice to have but i don't see it as a requirement, particularly when we have no control at the client end as per the win10 case

macosx however seems strange but i only have macosx 10.4 I could potentially test with at some point, what are the default perms macosx gives the share on mount?

**Comment by evan** [comment]
My Unraid SMB 3.x shares are discoverable in Windows 10 without legacy SMB support on either the client or the server, so there must be some way to get retronas to appear by default, but it's not a big deal to me. 

To my great surprise, inspecting the permissions of the folder fixes the issue on OS X until the next client reboot. No idea what's going on. Maybe it has something to do with crossing the filesystem boundary when the client first enumerates the permissions. 

<img width="656" alt="Screenshot 2023-08-19 at 5 36 05 PM" src="https://github.com/danmons/retronas/assets/210/9637de58-73f0-45d4-957c-395c5ce91c41">
<img width="372" alt="Screenshot 2023-08-19 at 5 33 26 PM" src="https://github.com/danmons/retronas/assets/210/cb342cb0-78de-403c-a94d-86e364da723e">
<img width="512" alt="Screenshot 2023-08-19 at 5 33 31 PM" src="https://github.com/danmons/retronas/assets/210/785951fc-ab00-4b13-a487-df25afbb9ebf">
<img width="657" alt="Screenshot 2023-08-19 at 5 36 16 PM" src="https://github.com/danmons/retronas/assets/210/00ae95a9-db01-4040-ac9f-3223a2958107">

**Comment by evan** [comment]
Maybe the way to fix it would be to have the etherdfs installer set up its own share instead of a folder. It could automatically use the msdos filesyatem type as well.

**Comment by sairuk** [comment]
you can test this theory if you like by switching to this branch

https://github.com/danmons/retronas/tree/etherdfs-samba

**Comment by sairuk** [comment]
i'm converting this to a discussion though because retronas is operating as expected

**Comment by evan** [comment]
I tried the branch; the dos samba share seems to be browsable without issues. The `dos` subfolder in the retronas share is still there as before.

**Reply by sairuk** [reply]
to confirm, this branch addresses the issues with macos access as per your initial suggestion?

**Reply by evan** [reply]
Yes it does, thank you. I find having the dos share also present as a directory in the main share confusing, but maybe that's just me. It seems like the PS2 share is also like that.

**Reply by sairuk** [reply]
the new share you are using is a share pointing at that folder under the main structure, we have no plans to change that approach

but if your initial suggestion works around the macos issue its a fair compromise thanks

**Reply by evan** [reply]
Thats fine. If there are any other mount points added within the main share in the future, they may have the same issue with OS X, but I really don't know.
